### PR TITLE
Fix cowHostList can't have hosts with same `ConnectAddress`

### PR DESCRIPTION
**Pull request by dkropachev**
Addresses [issue](https://github.com/gocql/gocql/issues/1757) from [gocql/gocql](https://github.com/gocql/gocql/)

Reopening of a https://github.com/scylladb/gocql/pull/182

### Problem

`cowHostList` uses `HostInfo.Equal` to confirm host uniqueness, which relies on `ConnectAddress.Equal`.
As result `cowHostList` does not allow to have different hosts with same `ConnectAddress`.

It breaks cases like accessing cluster via [tcp](https://github.com/gocql/gocql/issues/1757) or [ssl](https://github.com/gocql/gocql/issues/1452) proxy

### CODE REVIEW

File: host_source.go [hunk 1]
```diff
@@ -435,10 +435,10 @@ func (h *HostInfo) Hostname() string {
...
 
 func (h *HostInfo) ConnectAddressAndPort() string {
-        h.mu.Lock()
-        defer h.mu.Unlock()
-        addr, _ := h.connectAddressLocked()
-        return net.JoinHostPort(addr.String(), strconv.Itoa(h.port))
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	addr, _ := h.connectAddressLocked()
+	return net.JoinHostPort(addr.String(), strconv.Itoa(h.port))
```
Comment (sylwiaszunejko) [inline host_source.go:441]: These formatting changes are not related to this PR

File: policies_test.go [hunk 2]
```diff
@@ -132,47 +158,47 @@ func TestHostPolicy_TokenAware_LWT_DisablesHostShuffling(t *testing.T) {
...
+		}, routingKey: "8", lwt: true, shuffle: true, want: []string{"0", "2", "3", "4", "5", "1"}},
 		"token 08 shuffling not configured": {hosts: []*HostInfo{
 			{hostId: "0", connectAddress: net.IPv4(10, 0, 0, 1), tokens: []string{"00", "10", "20"}},
 			{hostId: "1", connectAddress: net.IPv4(10, 0, 0, 3), tokens: []string{"25", "35", "45"}},
 			{hostId: "2", connectAddress: net.IPv4(10, 0, 0, 2), tokens: []string{"00", "10", "20"}},
 			{hostId: "3", connectAddress: net.IPv4(10, 0, 0, 4), tokens: []string{"25", "35", "45"}},
 			{hostId: "4", connectAddress: net.IPv4(10, 0, 0, 3), tokens: []string{"50", "60", "70"}},
 			{hostId: "5", connectAddress: net.IPv4(10, 0, 0, 4), tokens: []string{"50", "60", "70"}},
-		}, routingKey: "8", lwt: true, shuffle: false, want: []string{"0", "2", "3", "1"}},
+		}, routingKey: "8", lwt: true, shuffle: true, want: []string{"0", "2", "3", "4", "5", "1"}},
```
Comment (sylwiaszunejko) [inline policies_test.go:169]: Why are you changing shuffling in this example?

**Review by sylwiaszunejko — CHANGES_REQUESTED** [review]

File: host_source.go (hunk 1 above)
Comment (dkropachev) [reply host_source.go:441]: Sure, but it is definitely formatting issues, here, for example, it had spaces instead of tabs, is there a way to get these type of changes in ?

File: policies_test.go (hunk 2 above)
Comment (dkropachev) [reply policies_test.go]: my fault, changed it back

File: host_source.go (hunk 1 above)
Comment (sylwiaszunejko) [reply host_source.go:441]: Separate PR I guess

Comment (dkropachev) [reply host_source.go]: Removed

**Review by sylwiaszunejko — APPROVED** [review]

**Comment by dkropachev** [comment]
@sylwiaszunejko , I can't make pipeline to pass, starting container is failing, it has nothing to do with this PR

**Comment by sylwiaszunejko** [comment]
@dkropachev please rebase to master and the pipeline should pass then
